code/modules/power/supermatter/supermatter_delamination.dm

```diff
@@ -1,13 +1,109 @@
 /datum/supermatter_delamination
+	///Power amount of the SM at the moment of death
 	var/supermatter_power = 0
+	///Amount of total gases interacting with the SM
+	var/supermatter_gas_amount = 0
+	///Base number of anomalies to spawn (can go up or down with a random small amount)
 	var/anomalies_to_spawn = 10
+	///Can we spawn anomalies after dealing with the delamination type?
+	var/should_spawn_anomalies = TRUE
+	///Reference to the supermatter turf
+	var/turf/supermatter_turf
+	///Baseline strenght of the explosion caused by the SM
+	var/supermatter_explosion_power = 0
+	///Amount the gasmix will affect the explosion size
+	var/supermatter_gasmix_power_ratio = 0
 
-/datum/supermatter_delamination/New(power)
+/datum/supermatter_delamination/New(power, combined_gas, turf/local_turf, explosion_power, gasmix_power_ratio, can_spawn_anomalies)
```

Comment:
The generally used style is to name all of these args the same as the vars they're called on `/datum/suermatter_delamination`
So you'd do like
```
/datum/supermatter_delamination/New(supermatter_power, turf/supermatter_turf, ...)
    if(supermatter_power)
        src.supermatter_power = supermatter_power
    if(supermatter_turf)
        src.supermatter_turf = supermatter_turf
```
etc for the rest 

See https://github.com/tgstation/tgstation/blob/master/.github/guides/STYLE.md#when-passing-vars-through-new-or-initializes-arguments-use-srcvar